src/components/PromotionalCard.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+
+import { useState, useEffect } from "react";
+
```

Comment:
We need to add this import so we can use the `<Image>` component (see the other comment).

```suggestion
import Image from "next/image";

```